src/modules/agent-network/AIProviderModal.tsx

```
@@ -337,23 +337,46 @@ export default function AIProviderModal({
+
+  // Auto-pick the first usable cluster on first render once the
   // /domains response lands. Only matters for the first-create flow —
   // once settings is bootstrapped no further bootstrap happens.
   React.useEffect(() => {
     if (settingsBootstrapped) return;
     if (bootstrapCluster) return;
-    if (validatedClusters.length === 0) return;
-    setBootstrapCluster(validatedClusters[0].domain);
-  }, [settingsBootstrapped, bootstrapCluster, validatedClusters]);
+    if (bootstrapClusters.length === 0) return;
+    setBootstrapCluster(bootstrapClusters[0].domain);
```

ESLint

[error] Error: Calling setState synchronously within an effect can trigger cascading renders

Effects are intended to synchronize state between React and external systems such as manually updating the DOM, state management libraries, or other platform APIs. In general, the body of an effect should do one or both of the following:
* Update external systems with the latest state from React.
* Subscribe for updates from some external system, calling setState in a callback function when external state changes.

Calling setState synchronously within an effect body causes cascading renders that can hurt performance, and is not recommended. (https://react.dev/learn/you-might-not-need-an-effect).

/home/jailuser/git/src/modules/agent-network/AIProviderModal.tsx:378:5
  376 |     if (bootstrapCluster) return;
  377 |     if (bootstrapClusters.length === 0) return;
> 378 |     setBootstrapCluster(bootstrapClusters[0].domain);
      |     ^^^^^^^^^^^^^^^^^^^ Avoid calling setState() directly within an effect
  379 |   }, [settingsBootstrapped, bootstrapCluster, bootstrapClusters]);
  380 |
  381 |   // Seed the upstream URL from the catalog entry once it lands — the

(react-hooks/set-state-in-effect)